crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -115,10 +115,13 @@ impl FormatRule<Expr, PyFormatContext<'_>> for FormatExpr {
             Parentheses::Never => false,
         };
         if parenthesize {
-            let comment = f.context().comments().clone();
-            let node_comments = comment.leading_dangling_trailing(expression);
+            let comments = f.context().comments().clone();
+            let node_comments = comments.leading_dangling_trailing(expression);
             if !node_comments.has_leading() && !node_comments.has_trailing() {
-                parenthesized("(", &format_expr, ")").fmt(f)
+                let hug = f.options().preview().is_enabled() && is_expression_huggable(expression);
+                parenthesized("(", &format_expr, ")")
+                    .with_indent(!hug)
+                    .fmt(f)
             } else {
                 format_with_parentheses_comments(expression, &node_comments, f)
```

Comment:
Do your tests cover the `format_with_parentheses_comments` case? If not, could you add tests for it as well?

Reply:
Yes, I believe this path isn't necessary since we don't want to hug when the expression has leading or trailing comments.